scanner/scanner.coffee

```diff
@@ -11,62 +11,76 @@ error = require '../error/error.coffee'
 
 LETTER = XRegExp '[\\p{L}]'
 DIGIT = XRegExp '[\\p{Nd}]'
+QUOTES = XRegExp '\"|\''
 WORD_CHAR = XRegExp '[\\p{L}\\p{Nd}_]'
 KEYWORDS = ///
   (make|to|deeg|end|thru|till|by|exists|and|or|unless|if|else|then|not|
   true|false|for|while|does|count|counts|match|with)
 ///
 
 module.exports = (filename, callback) ->
-  scanningError = null
+  scanningError = []
+  tokens = []
   baseStream = fs.createReadStream filename, {encoding: 'utf8'}
-  baseStream.on 'error', (err) -> console.log error err
+  baseStream.on 'error', (err) ->
+    scanningError.push error err
+    callback scanningError, tokens
 
   stream = byline baseStream, {keepEmptyLines: true}
-  tokens = []
   lineNumber = 0
   blockComment = {
     yes: false
   }
   stream.on 'readable', () ->
-    scan stream.read(), ++lineNumber, tokens, blockComment
+    scan stream.read(), ++lineNumber, tokens, blockComment, scanningError
   stream.once 'end', () ->
     tokens.push {kind: 'EOF', lexeme: 'EOF'}
     callback scanningError, tokens
     
-scan = (line, lineNumber, tokens, blockComment) ->
+scan = (line, lineNumber, tokens, blockComment, scanningError) ->
   return if not line
 
   [start, pos] = [0, 0]
 
   emit = (kind, lexeme) ->
     tokens.push {kind, lexeme: lexeme or kind, line: lineNumber, col: start+1}
 
-  loop
+  toHex = (num) ->
```

Comment:
`toHex` could be converted to `num.toString(16)` which will convert an `int` into a `string` of hex value.